cli/dcoscli/auth/main.py

```diff
@@ -68,34 +75,139 @@ def _info(info):
     return 0
 
 
-def _login():
+def _list_providers(json_):
     """
-    :returns: process status
+    :returns: providers available for configured cluster
+    :rtype: dict
+    """
+
+    providers = auth.get_providers()
+    if providers or json_:
+        emitting.publish_table(
+            emitter, providers, tables.auth_provider_table, json_)
+    else:
+        raise DCOSException("No providers configured for your cluster")
+
+
+def _get_password(password_str, password_env, password_file):
+    """
+    Get password for authentication
+
+    :param password_str: password
+    :type password_str: str
+    :param password_env: name of environment variable with password
+    :type password_env: str
+    :param password_file: path to file with password
+    :type password_file: bool
+    :returns: password or None if no password specified
+    :rtype: str | None
+    """
+
+    password = None
+    if password_str:
+        password = password_str
+    elif password_env:
+        password = os.environ.get(password_env)
+    elif password_file:
+        password = util.read_file(password_file)
```

Comment:
We should validate the permissions of that file and error out if they are too open. Do we already do that? See my corresponding comment further below, in the context of the service account login, where I propose something in the lines of `util.read_file_secure(path)`.